cmake/modules/SwiftComponents.cmake

```diff
@@ -92,11 +92,21 @@ macro(swift_configure_components)
     "A semicolon-separated list of components to install from the set ${_SWIFT_DEFINED_COMPONENTS}")
 
   foreach(component ${_SWIFT_DEFINED_COMPONENTS})
+    add_custom_target(${component})
+    add_swift_install_targets(${component} ${component})
+
     string(TOUPPER "${component}" var_name_piece)
     string(REPLACE "-" "_" var_name_piece "${var_name_piece}")
     set(SWIFT_INSTALL_${var_name_piece} FALSE)
   endforeach()
 
+  # NOTE: never-install is a dummy component to indicate something should not
+  # be installed. We explicitly do not add an install target for this.
+  add_custom_target(never_install)
```

Comment:
Bleh, I think that we should remove the `never_install` target instead.

Reply:
I had a patch to make `INSTALL_IN_COMPONENT` optional, but I don't think @Rostepher liked it too much. 😛